### ISSUE TITLE
`brace-style`: Option for styles to be enforced on  curly brackets around `case` statements

### ISSUE THREAD

**Issue by brettz9** [eslint/eslint#13331]
**What rule do you want to change?**

- `brace-style`

**Does this change cause the rule to produce more or fewer warnings?**

More warnings, though in the case of the initial bracket of the `switch` statement with `1tbs`, it could also end up producing fewer (see comment within the `1tbs` example below).

**How will the change be implemented? (New option, new default behavior, etc.)?**

New option (or possibly options) to avoid breaking changes though for a new major release, accepting it as the new default would, imo, be reasonable here.

**Please provide some example code that this change will affect:**

I think this is easiest to highlight exclusively through passing examples, noting that if any of the string option values are changed to any of the other values (e.g., "1tbs" to "allman"), the examples will then instead fail.

```js
/* eslint brace-style: ["error", "1tbs", {applyToSwitchCases: true}] */
switch (sth) 
// For this style, the first curly brace (actually belonging to the `switch`) might
//  be seen as needing to go above (as is the current expectation with this rule
//  using this "1tbs" option--a rule which does enforce for `switch`, just not for
//  `case`), but as it helps with alignment when `case` statements have their
//  own braces, I think it is much better aligned this way (regardless, it'd be nice
//  to enforce consistency).
// If considered useful, perhaps another option could be added on whether to
//  ignore, or enforce a specific placement of this initial `switch` brace (whether
//  providing different behavior depending on whether curly brackets were in
//  use or not)
{ case 1: {

} case 2: {

}
}
```

```js
/* eslint brace-style: ["error", "stroustrup", {applyToSwitchCases: true}] */
switch (sth) {
case 1: {

}
case 2: {

}
}
```

```js
/* eslint brace-style: ["error", "allman", {applyToSwitchCases: true}] */
switch (sth) 
{
case 1:
{

}
case 2:
{

}
}
```

If it were deemed useful, there could also be a `applyToSwitchCasesOnly` option to allow those wishing to enforce style on cases but not `if`/`else`, etc..

And if some may wish for a different style to be enforced for `case`'s, e.g., `1tbs` for most control statements, but `stroustrup` for `case` statements, the option could instead become a string, e.g., `switchCaseStyle: 'stroustrup'`.

**What does the rule currently do for this code?**

Ignores reporting it regardless of style (except for the first bracket in the `1tbs` style; see the code comment in that example).

**What will the rule do after it's changed?**

Report when the `case` block's violate the style.

**Are you willing to submit a pull request to implement this change?**

No.

I think this rule has become more of a desirable feature to enforce given the increasing use of `const` and `let` (aided by the `no-var` rule), and particularly when the recommended [no-case-declarations](https://eslint.org/docs/rules/no-case-declarations) is being taken advantage of.

**Comment by anikethsaha** [comment]
Though this enhancement is doable. 

One thing I think can be conflict/inconsistent is the indentation for the `case` blocks. usually they are one level more indented than the outer block (`switch`) so may be style like `allman` may be affect that. 

let me know if I missed something about this.

**Comment by kaicataldo** [comment]
The team has made the difficult decision to freeze stylistic rules (please see https://eslint.org/blog/2020/05/changes-to-rules-policies for more details), and we unfortunately won't be able to accept this proposal.

Thanks for contributing to ESLint!
